### PR TITLE
Replace BooleanInstances with Boolean companion object

### LINKED ISSUE #943: Replace BooleanInstances with Boolean companion object
Starting in Kotlin 1.3, Boolean now has a companion object: https://youtrack.jetbrains.com/issue/KT-7922

We can start the deprecation process for BooleanInstances:

https://github.com/arrow-kt/arrow/blob/master/modules/core/arrow-core/src/main/kotlin/arrow/core/boolean.kt#L3

https://github.com/arrow-kt/arrow/blob/master/modules/core/arrow-instances-core/src/main/kotlin/arrow/instances/boolean.kt#L16-L27

Steps: 

- [ ] Add @deprecated flag today
- [ ] Move methods to Boolean companion when 1.3 is released

### PR DESCRIPTION
**Pull request by starkej2**
Should resolve #943

### CODE REVIEW

**Comment by raulraja** [comment]
@starkej2 thanks for the fix. For this to work and compile I think we have to bump the Kotlin version. Arrow 0.8.0 is meant to be released alongside Kotlin 1.3. I'll keep this unmerged until we are ready to bump the Kotlin version unless you want to attempt that yourself.

You can verify Arrow builds correctly by simply importing it from IntelliJ as an external project and perform a full project build. 

In order to verify the entire project you have to use the command line with:
```groovy
./gradlew clean build :arrow-docs:runAnk
```

File: modules/core/arrow-instances-core/src/main/kotlin/arrow/instances/boolean.kt [hunk 1]
```diff
@@ -1,6 +1,5 @@
 package arrow.instances
 
-import arrow.core.BooleanInstances
 import arrow.core.DeprecatedAmbiguity
```
Comment (pakoito) [inline modules/core/arrow-instances-core/src/main/kotlin/arrow/instances/boolean.kt]: This import is not used anymore, it can be removed.

**Comment by starkej2** [comment]
I'm getting a build error, but it appears to also happen on the `paco-kotlin1.3 branch`

```
> Task :arrow-docs:runAnk FAILED
Exception in thread "main" java.lang.ClassCastException: arrow.typeclasses.MonadKt$binding$wrapReturn$1 cannot be cast to kotlin.coroutines.experimental.Continuation
  at arrow.ank.AnkKt$ank$1.invoke(ank.kt)
  at arrow.typeclasses.MonadKt$binding$wrapReturn$1.invokeSuspend(Monad.kt:54)
  at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:32)
  at kotlin.coroutines.ContinuationKt.startCoroutine(Continuation.kt:128)
  at arrow.typeclasses.MonadKt.binding(Monad.kt:55)
  at arrow.ank.AnkKt.ank(ank.kt:15)
  at arrow.ank.main.main(main.kt:17)

FAILURE: Build failed with an exception.
```

**Comment by pakoito** [comment]
@starkej2 do you mind reopening this PR against master, or can we do it? We need it for the 0.8.0 release now that 1.3 is on the way!

**Comment by starkej2** [comment]
Sure, no problem. I can do that today.

**Comment by starkej2** [comment]
Rebased onto master @pakoito

**Comment by pakoito** [comment]
Thank you @starkej2 !! We can go ahead with 0.8.0 over the weekend :D Thanks for your first contribution

**Comment by starkej2** [comment]
> Thanks for your first contribution

Hopefully the first of many! Thanks for the help.

**Pull request merged**